### PR TITLE
Ai mode

### PR DESCRIPTION
**Pull request by nastiche**
Please review this PR from the very beginning (first review happened a few days ago and after that I refactored the whole code and added lots of comments to make it readable).

Contact me please in case you still have difficulties reading the code.

### CODE REVIEW

File: pages/api/tasks/ai/index.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    const aiTaskData = {
+      title: object.Title,
+      subtasks: object.Subtasks,
+      deadline: null,
+      tags: [],
+      priority: "",
+      originalTaskDescription: query,
+    };
+    console.log(aiTaskData);
```
Comment (dimitriosxmi) [inline pages/api/tasks/ai/index.js:66]: `console.log()`

File: pages/create.js [hunk 2]
```diff
@@ -1,24 +1,137 @@
...
+          });
+          setAiMode(true);
+        }
+      } catch (error) {
+        console.error("Error:", error);
+      }
     }
+    setIsLoading(false);
   }
+  console.log(aiTaskDescription);
```
Comment (dimitriosxmi) [inline pages/create.js:91]: `console.log()`

File: pages/api/tasks/index.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ export default async function handler(request, response) {
     try {
       const taskData = request.body;
       await Task.create(taskData);
+      await
```
[the remark this answers is not in the document]
Comment (nastiche) [reply pages/api/tasks/index.js:11]: der Rest des codes ist im PR nicht zu sehen, sonst macht es Sinn:

if (request.method === "POST") {
    try {
      const taskData = request.body;
      await Task.create(taskData);
      await response.status(201).json({ status: "Task created" });
    } catch (error) {
      console.log(error);
      response.status(400).json({ error: error.message });
    }
  }

File: pages/api/tasks/ai/index.js (hunk 1 above)
Comment (nastiche) [reply pages/api/tasks/ai/index.js:63]: will fix

File: pages/create.js (hunk 2 above)
Comment (nastiche) [reply pages/create.js:91]: will fix

**Review by actyvystom — CHANGES_REQUESTED** [review]
Amazing loads of code lines! 🤯 
I picked a few simple issues but all in all these are far too many lines of code to get an overview in a reasonable time - although very nicely commented.  
As this is refactoring, I suggest to break it down into edible pieces (PRs).
Cheers, Thomas

**Comment by CarstenJahnke** [comment]
Wooh, so much changes 😝🚀 LGTM

**Pull request merged**